### PR TITLE
added a validation module to validate inputs and positions

### PR DESCRIPTION
**Pull request by GASANAMDavid**
**Major changes**

> Tested and added validation module which responds with appropriate error messages if any.

### CODE REVIEW

File: lib/David_tic_tac_toe.rb [hunk 1]
```diff
@@ -13,10 +14,14 @@ def self.play
     while true
       game.game_board.print_board
       puts "------------------\n\nPlayer #{current_player.name}'s turn: "
+      puts 'Enter indices separated by  a comma'
       input = gets.chomp
-      indices = input.split(' ', 2)
+      next unless Validation.validate_input(input)
```
Comment (GASANAMDavid) [inline lib/David_tic_tac_toe.rb:19]: > Validates if user input can be split into only 2 values using a comma as a separator

> valid if the user input is free from non-numeric characters.

File: lib/David_tic_tac_toe/validation.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+module Validation
```
Comment (GASANAMDavid) [inline lib/David_tic_tac_toe/validation.rb:2]: This module handles all validation s

File: spec/David_tic_tac_toe/validation_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+require_relative '../../lib/David_tic_tac_toe/validation'
```
Comment (GASANAMDavid) [inline spec/David_tic_tac_toe/validation_spec.rb:1]: Tests for validation module

File: lib/David_tic_tac_toe.rb [hunk 4]
```diff
@@ -13,10 +14,14 @@ def self.play
...
+
+      indices = input.split(',', 2)
       indices = indices.map(&:to_i)
```
Comment (GASANAMDavid) [inline lib/David_tic_tac_toe.rb:22]: Changed separator from a space to a comma

File: lib/David_tic_tac_toe.rb [hunk 5]
```diff
@@ -13,10 +14,14 @@ def self.play
...
-      next unless game.game_board.valid_position(indices) || game.game_board.board_full
+      # puts indices
```
Comment (naush) [inline lib/David_tic_tac_toe.rb]: I would remove the debug statements even if they are commented out so that there is less noise in your code.

File: lib/David_tic_tac_toe/validation.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  def self.validate_input(input)
+    input.gsub!(/[[:space:]]/, '')
+    user_input = input.split(',')
+    (user_input.length == 2) && user_input.all?(/[0-9]/)
+  end
+
+  def self.validate_position(index, board)
+    puts index
+    puts board
```
Comment (naush) [inline lib/David_tic_tac_toe/validation.rb]: I would remove these debug statements well.

File: lib/David_tic_tac_toe/validation.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    denote_empty = '_'
```
Comment (naush) [inline lib/David_tic_tac_toe/validation.rb]: Since this is already defined in the Board class, I would try to reference the constant from here instead of creating another one.

File: spec/David_tic_tac_toe/validation_spec.rb [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    let(:index) { [4, 3] }
+    let(:board) { [%w[_ _ _], %w[_ _ _], %w[X _ _]] }
+    it "should return false if index is out of range" do
+      expect(subject.validate_position(index, board)).to eq false
+    end
+
+    it "should return false if index was already played" do
+      expect(subject.validate_position([2, 0], board)).to eq false
+    end
+  end
```
Comment (naush) [inline spec/David_tic_tac_toe/validation_spec.rb:37]: Remember to test `happy path` as well as the edge cases.

File: spec/David_tic_tac_toe/validation_spec.rb [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+RSpec.describe Validation do
+  let(:input) { '12 , 12,a' }
+  it "should return false if input is not separated by a single comma" do
+    expect(subject.validate_input(input)).to eq false
+  end
+
+  it "should return false if input contains a non numeric character" do
+    expect(subject.validate_input('12,a')).to eq false
+  end
```
Comment (naush) [inline spec/David_tic_tac_toe/validation_spec.rb:21]: I would have these two test cases in a context under `.validate_input`.

Comment (naush) [reply spec/David_tic_tac_toe/validation_spec.rb:21]: It looks like you are missing test cases for `input.gsub!(/[[:space:]]/, '')`. If you don't have a test for it, then we don't know if the code works!

**Comment by naush** [comment]
Good effort! 💯 

After you are finish with adding validation, I'd like to see you work on applying Law of Demeter and Single Responsibility Principle!

File: lib/David_tic_tac_toe.rb (hunk 5 above)
Comment (GASANAMDavid) [reply lib/David_tic_tac_toe.rb]: 1b8092c74aedfd9c1ae34fa91dee6692b08e5eb5

File: lib/David_tic_tac_toe/validation.rb (hunk 6 above)
Comment (GASANAMDavid) [reply lib/David_tic_tac_toe/validation.rb]: 1b8092c74aedfd9c1ae34fa91dee6692b08e5eb5

File: lib/David_tic_tac_toe/validation.rb (hunk 7 above)
Comment (GASANAMDavid) [reply lib/David_tic_tac_toe/validation.rb]: 1b8092c74aedfd9c1ae34fa91dee6692b08e5eb5

File: spec/David_tic_tac_toe/validation_spec.rb (hunk 9 above)
Comment (GASANAMDavid) [reply spec/David_tic_tac_toe/validation_spec.rb:21]: > I would have these two test cases in a context under `.validate_input`.

1b8092c74aedfd9c1ae34fa91dee6692b08e5eb5

File: spec/David_tic_tac_toe/validation_spec.rb (hunk 8 above)
Comment (GASANAMDavid) [reply spec/David_tic_tac_toe/validation_spec.rb:37]: 1b8092c74aedfd9c1ae34fa91dee6692b08e5eb5
